WorkloadManagementSystem/PilotAgent/pilotCommands.py

```diff
@@ -359,6 +358,42 @@ def _getSecurityCFG( self ):
       self.cfg.append( "-o /DIRAC/Security/CertFile=%s/hostcert.pem" % self.pp.certsLocation )
       self.cfg.append( "-o /DIRAC/Security/KeyFile=%s/hostkey.pem" % self.pp.certsLocation )
 
+class CheckCECapabilities( CommandBase ):
+  """ Used to get  CE tags.
+  """
+  def __init__( self, pilotParams ):
+    """ c'tor
+    """
+    super( CheckCECapabilities, self ).__init__( pilotParams )
+    self.cfg = []
+
+
+  def execute( self ):
+
+    retCode, resourceDict = self.executeAndGetOutput( 'dirac-resource-get-parameters -S %s -N %s -Q %s' %
```

Comment:
  Hi Federico, Cinzia,
      
        I see, thanks for having tested it. So, let's continue with the
      current implementation by calling the
      external command. We have to still think how the usual DIRAC
      modules can be used once it is
      installed. This is not trivial indeed but can be quite beneficial.
      
        Cheers,
        Andrei 
      
      On 26/10/15 13:00, fstagni wrote:
    
    
      In WorkloadManagementSystem/PilotAgent/pilotCommands.py:
      > @@ -359,6 +358,42 @@ def _getSecurityCFG( self ):
>        self.cfg.append( "-o /DIRAC/Security/CertFile=%s/hostcert.pem" % self.pp.certsLocation )
>        self.cfg.append( "-o /DIRAC/Security/KeyFile=%s/hostkey.pem" % self.pp.certsLocation )
>  
> +class CheckCECapabilities( CommandBase ):
> +  """ Used to get  CE tags.
> +  """
> +  def __init__( self, pilotParams ):
> +    """ c'tor
> +    """
> +    super( CheckCECapabilities, self ).__init__( pilotParams )
> +    self.cfg = []
> +
> +
> +  def execute( self ):
> +
> +    retCode, resourceDict = self.executeAndGetOutput( 'dirac-resource-get-parameters -S %s -N %s -Q %s' %

      So, after test: we can't use gConfig here, and this is clear as
        soon as we do "from DIRAC import gConfig" or any other "from
        DIRAC* import *" in any part of the pilot code. 2 reasons:
      
        if (VO)DIRAC is installed with dirac-install, it will still
          complain that can't find GSI. And this is because the
          environment at this point is not set (it's not in the current
          process)
        if (VO)DIRAC is installed from, say, CVMFS, still the
          environment created has to be passed in to the subprocess that
          run the scripts.
      
      —
        Reply to this email directly or view
          it on GitHub.